# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/BrightTimothy/FlyRank-AI-Machine-Learning-Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

I'm picking Lane 4: CTR/Engagement Opportunity Scoring

Search visibility alone doesn't guarantee traffic, a page can rank well and still under capture clicks or engagement relative to its position. This lane lets me find those pages specifically, using position adjusted CTR comparisons rather than naive click counts, which produces more actionable, less noisy recommendations than a general ranking-signal study would.

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

Decision: Which visible pages (already ranking, already getting impressions) are under capturing clicks or engagement relative to peers in the same position tier, and deserve a title/meta/content review?

Who acts on it: A content reviewer or SEO strategist working through a prioritized review queue with limited time, they can only manually check a handful of pages per week, not the full inventory.

Cost of a wrong call: If I flag a page as an opportunity and it isn't one, the reviewer wastes time on a rewrite that won't move the needle. If I miss a real opportunity, a page keeps quietly under performing earning impressions it can't convert into clicks, while a genuinely fixable problem goes untouched.

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
import pandas as pd

df = pd.read_csv("https://raw.githubusercontent.com/BrightTimothy/FlyRank-AI-Machine-Learning-Internship/main/data/raw/content_refresh_anonymized.csv")

# Reason code: low_ctr_visible_page
# impressions_90d >= 500, avg_position between 1-20, ctr < 0.5
low_ctr = df[
    (df["impressions_90d"] >= 500) &
    (df["avg_position"] > 0) & (df["avg_position"] <= 20) &
    (df["ctr"] < 0.5)
]
print("Low-CTR visible pages:", len(low_ctr), f"({len(low_ctr)/len(df)*100:.1f}% of dataset)")

# Reason code: low_engagement_visible_page
# sessions_90d >= 30, engagement_rate or scroll_rate below 30
low_engagement = df[
    (df["sessions_90d"] >= 30) &
    ((df["engagement_rate"] < 30) | (df["scroll_rate"] < 30))
]
print("Low-engagement visible pages:", len(low_engagement))

# CTR by position tier — shows why position-adjustment matters
print(df.groupby("position_tier")["ctr"].mean().sort_index())


Low-CTR visible pages: 9759 (32.5% of dataset)
Low-engagement visible pages: 7113
position_tier
deep        0.150212
page_1      0.652467
page_3_5    0.222484
striking    0.323239
top_3       1.483611
Name: ctr, dtype: float64


## 4. Careful words: what I can and can't claim

I can say: In this starter slice, 32.5% of visible pages (9,759 of them) meet the low-CTR reason-code criteria, and 7,113 pages meet the low-engagement criteria — both sizeable enough to be worth a 7-week investigation, not noise. CTR also clearly differs by position tier (from ~0.15 in the deepest tier to ~1.48 in the top-3 tier), which confirms the core assumption behind this lane: comparing raw CTR without adjusting for position would be misleading, since a "low" CTR at position 1 means something very different from a "low" CTR buried on page 3. These are observed, directional patterns — a starting point for a prioritized review queue, not a verdict on any individual page.

I can't say: that a flagged page's low CTR is caused by anything specific (a weak title, a bad meta description, wrong intent match) — I haven't tested any of those explanations, only observed the outcome. I can't claim these numbers hold beyond this specific 30,000-row anonymized starter slice — the full warehouse could look different. I can't claim that fixing a flagged page will improve its CTR, since that would require an actual experiment (a before/after test), not just this observational data. And I can't make any claim about how Google's ranking algorithm works — CTR is a measurable outcome, not a window into search engine internals.

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.